ocaml/xapi/xapi_vm.ml

```diff
@@ -1007,12 +1008,15 @@ let set_snapshot_schedule ~__context ~self ~value =
 		if Db.VM.get_is_a_template ~__context ~self then
 			(* Do not assign templates to a VMSS. *)
 			raise (Api_errors.Server_error(Api_errors.vm_is_template, [Ref.string_of self]));
-		(* Check VM allowed operation support the snapshot type of VMSS *)
-		let allowed_operations = Db.VM.get_allowed_operations ~__context ~self in
+		(* Check VM allowed operations for VMSS snapshot_type=snapshot_with_quiesce *)
 		let snapshot_type = Db.VMSS.get_type ~__context ~self:value in
-		if not (List.exists (fun ty -> ty == snapshot_type) allowed_operations) then
-			raise (Api_errors.Server_error(Api_errors.operation_not_allowed,
-				["VM doesn't support snapshot_type " ^ (Record_util.vmss_type_to_string snapshot_type)]));
+		if snapshot_type = `snapshot_with_quiesce then begin
+			Pool_features.assert_enabled ~__context ~f:Features.VSS;
+			let allowed_operations = Db.VM.get_allowed_operations ~__context ~self in
+			if not (List.exists (fun ty -> ty = snapshot_type) allowed_operations) then
```

Comment:
Is there a special reason to use `List.exists`? Otherwise I'd write:

```
if not (List.mem snapshot_type allowed_operations) then
```
or
```
if not (List.mem snapshot_type (Db.VM.get_allowed_operations ~__context ~self)) then
   raise ..
```

We already know that `snapshot_type` is `snapshot_with_quiesce` but I'd keep using `snapshot_type`.